opennlp-tools/src/test/java/opennlp/tools/util/normalizer/UrlCharSequenceNormalizerTest.java

```diff
@@ -44,4 +44,15 @@ public void normalizeEmail() throws Exception {
         "asdf   2nnfdf  ", normalizer.normalize("asdf asd.fdfa@hasdk23.com.br" +
             " 2nnfdf asd.fdfa@hasdk23.com.br"));
   }
+
```

Comment:
I don't like this test because it relies on timing.  I can remove it or substitute something better if you have recommendations.

Reply:
Why does it rely on timing?

Reply:
I couldn't think of another way to test that a regex was thrashing and taking a long, long time to process.  Any recommendations?

Reply:
I think this is a good change but I do worry about limiting the length of the URL in the regex. What if we added an argument to the `UrlCharSequenceNormalizer` constructor to make this an option to the user? That way the user can choose between the trade off of speed vs. potentially missing URLs and there won't be any risk of changing the expected behavior of OpenNLP language detector applications out in the wild. Thoughts?